src/test/java/seedu/todo/logic/commands/CommandPreviewTest.java

```diff
@@ -0,0 +1,68 @@
+package seedu.todo.logic.commands;
+
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+
+import static org.junit.Assert.*;
+import static org.junit.matchers.JUnitMatchers.*;
+import static org.hamcrest.CoreMatchers.*;
+import static org.mockito.Mockito.*;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.Before;
+import org.junit.Rule;
+
+//@@author A0139021U
+public class CommandPreviewTest {
+
+    @Rule
+    public MockitoRule rule = MockitoJUnit.rule();
+    
+    @Before
+    public void setUpPreview() throws Exception {
+        Set<String> mockCommands = Sets.newHashSet("add", "delete");
+    }
+    
+    @Test
+    public void testFilterAdd() throws Exception {
+        // TODO: find way to mock static methods
+        List<CommandSummary> expected = CommandMap.getCommand("add").getCommandSummary();
+        List<CommandSummary> actual = new CommandPreview("add").getPreview();
+        assertTrue(isShallowCompareCommandSummaries(expected, actual));
```

Comment:
Instead of this, place the asserts in the function itself, otherwise this will provide very little useful information when the test fails